torch_geometric/nn/aggr/basic.py

```diff
@@ -66,6 +66,130 @@ def forward(self, x: Tensor, index: Optional[Tensor] = None,
+    """
+
+    interpolations = {'linear', 'lower', 'higher', 'nearest', 'midpoint'}
+
+    def __init__(self, q: float, interpolation: str = 'linear',
+                 fill_value: float = float('nan')):
+        if q > 1. or q < 0:
+            raise ValueError("'q' must be in the range [0, 1].")
+        if interpolation not in self.interpolations:
+            raise ValueError(f"Invalid '{interpolation}' interpolation method."
+                             f" Available interpolations:"
+                             f"{self.interpolations}")
+
+        super().__init__()
+        self.q = q
+        self.interpolation = interpolation
+        self.fill_value = fill_value
+
+    def forward(self, x: Tensor, index: Optional[Tensor] = None,
+                ptr: Optional[Tensor] = None, dim_size: Optional[int] = None,
+                dim: int = -2) -> Tensor:
+        # we could also expand `ptr` to an index
+        self.assert_index_present(index)
+
+        if dim_size is None:
+            dim_size = index.max() + 1
+
+        # compute the quantile points
+        count = torch.bincount(index, minlength=dim_size)
+        cumsum = torch.cumsum(count, dim=0)
+        q_point = self.q * (count - 1) + cumsum - count
+
+        # shape used for expansions
+        # (1, ..., 1, -1, 1, ..., 1)
+        shape = [1] * x.dim()
+        shape[dim] = -1
+
+        # two sorts: the first one on the value,
+        # the second (stable) on the indices
+        x, x_perm = torch.sort(x, dim=dim)
+        index = index.view(*shape).expand_as(x)
+        index = index.take_along_dim(x_perm, dim=dim)
+        index, index_perm = torch.sort(index, dim=dim, stable=True)
+        x = x.take_along_dim(index_perm, dim=dim)
+
+        # compute the quantile interpolations
+        if self.interpolation == 'lower':
+            idx = torch.floor(q_point).long()
+            median = x.index_select(dim, idx)
+        elif self.interpolation == 'higher':
+            idx = torch.ceil(q_point).long()
+            median = x.index_select(dim, idx)
+        elif self.interpolation == 'nearest':
+            idx = torch.round(q_point).long()
+            median = x.index_select(dim, idx)
+        else:
+            l_idx = torch.floor(q_point).long()
+            r_idx = torch.ceil(q_point).long()
+            l_med = x.index_select(dim, l_idx)
+            r_med = x.index_select(dim, r_idx)
+
+            if self.interpolation == 'linear':
+                q_frac = torch.frac(q_point).view(*shape)
+                median = l_med + (r_med - l_med) * q_frac
+            else:  # 'midpoint'
+                median = 0.5 * l_med + 0.5 * r_med
+
+        # if the number of elements is 0, return 'nan'/fill_value
+        out_mask = (count > 0).view(*shape)
+        return torch.where(
+            out_mask, median,
+            torch.tensor([self.fill_value], dtype=median.dtype,
+                         device=median.device))
+
+    def __repr__(self) -> str:
```

Comment:
We do have test cases  for most of the custom representations

Reply:
I'll make one also for this!